npm/private/pkg_glob.bzl

```diff
@@ -0,0 +1,72 @@
+"""
+Basic glob implementation required for the pnpm public-hoist-exp option
+(https://pnpm.io/npmrc#public-hoist-exp).
+
+pnpm implementation and tests:
+    https://github.com/pnpm/pnpm/blob/v7.4.0-2/packages/matcher/src/index.ts
+    https://github.com/pnpm/pnpm/blob/v7.4.0-2/packages/matcher/test/index.ts
+"""
+
+def pkg_glob(exp, pkg):
+    """Testing if the passed pkg matches the exp.
+
+    Args:
+        exp: the glob expression
+        pkg: the package name to test
+
+    Returns:
+        True if the package matches the glob expression
+    """
+    exp_i = 0
+    pkg_i = 0
+
+    # HACK: assume "|" will never be in the expression and use it to
+    # split on * while keeping * as an array entry.
+    exp_parts = exp.replace("*", "|*|").strip("|").split("|")
```

Comment:
hmm. this might blow up if a user passes "**"; should probably fail if there is a "**" and say that this isn't valid syntax for this function